src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
```

Comment:
PullProofAI

There is a potential issue with accessing properties on the `err` object. It's better to perform null checks before accessing nested properties to avoid runtime errors.

```diff
-     err = err.response ? err.response : JSON.stringify(err);
+     err = err && err.response ? err.response : JSON.stringify(err);
```



<!-- This is an auto-generated comment by PullProofAI -->